app/javascript/packs/admin_shifts.js

```diff
@@ -133,13 +166,77 @@ fetch("/admin/shifts/get_external_staff_needed", {
     calendar.render();
   });
 
-let createEvent = (arg = undefined) => {
-  openModal(arg);
+// Refresh pending Shifts
+const refreshPendingShifts = () => {
+  fetch("/admin/shifts/pending_shifts?format=js", {
+    method: "GET",
+  })
+    .then((r) => r.text())
+    .then((html) => {
+      const fragment = document.createRange().createContextualFragment(html);
+      document
+        .getElementById("pending-shift-partial")
+        .replaceChildren(fragment);
+    });
+};
+
+const populateUsers = (arg) => {
+  userIdInput.tomselect.clearOptions();
+  let startDate, endDate;
+  if (arg.event) {
+    startDate = arg.event.start;
+    endDate = arg.event.end;
+  } else {
+    startDate = arg.start;
+    endDate = arg.end;
+  }
+  let startHour = startDate.toUTCString().split(" ")[4].split(":")[0];
+  let startMinute = startDate.toUTCString().split(" ")[4].split(":")[1];
+  let endHour = endDate.toUTCString().split(" ")[4].split(":")[0];
+  let endMinute = endDate.toUTCString().split(" ")[4].split(":")[1];
+  let weekDayInt = {
+    "Sun,": 0,
+    "Mon,": 1,
+    "Tue,": 2,
+    "Wed,": 3,
+    "Thu,": 4,
+    "Fri,": 5,
+    "Sat,": 6,
+  }[startDate.toUTCString().split(" ")[0]];
+  fetch(
```

Comment:
I think it would be cleaner using `/admin/shifts/shift_suggestions?start=${startHour}:${startMinute}&start=${endHour}:${endMinute}&day=${weekDayInt}`